bluebrain/repo-bluebrain/packages/neurodamus-models/package.py

```diff
@@ -72,10 +72,25 @@ def setup_run_environment(self, env):
         env.set(f"NEURODAMUS_{mech.upper()}_ROOT", self.prefix)
         env.set("HOC_LIBRARY_PATH", self.prefix.share.join(f"neurodamus_{mech}").hoc)
         if "+coreneuron" in self.spec:
-            env.set("CORENEURONLIB", self.prefix.lib + "/libcorenrnmech.so")
-        for libnrnmech_name in find(self.prefix.lib, "libnrnmech*.so", recursive=False):
-            env.set("NRNMECH_LIB_PATH", libnrnmech_name)
-            env.set("BLUECELLULAB_MOD_LIBRARY_PATH", libnrnmech_name)
+            import os
+
+            if os.path.exists(self.prefix.lib + "/libcorenrnmech.so"):
+                env.set("CORENEURONLIB", self.prefix.lib + "/libcorenrnmech.so")
+            elif os.path.exists(self.prefix.lib + "/libcorenrnmech.dylib"):
+                env.set("CORENEURONLIB", self.prefix.lib + "/libcorenrnmech.dylib")
+            else:
+                Exception("No CORENEURONLIB found!")
+
+        libnrnmech_name = find(self.prefix.lib, "libnrnmech*.so", recursive=False) + find(
```

Comment:
Maybe `or` instead of `+` here ?

Reply:
because it creates a list and I would like to throw an error if I find multiple libs: something fishy is going on

Reply:
So you will always look for `libnrnmech*.so` and `libnrnmech*.dylib`? I am thinking of searching for `.dylib` in case of no `.so` because I don't think they would exist at the same time. The exception would be thrown when multiple libraries are found in either case.

Reply:
yes this is the only main difference. I do not think calling another find really slows down the code in any significative way (also in big-O notation it is the same because we are already calling 1 find). Anyway, I do not have strong opinions on this. If you prefer the other way, I will change

Reply:
Hmm.. Then I'd prefer `or`.  Thanks!